kernel-spark/src/main/java/io/delta/kernel/spark/read/SparkScanBuilder.java

```diff
@@ -34,16 +35,20 @@
  * A Spark ScanBuilder implementation that wraps Delta Kernel's ScanBuilder. This allows Spark to
  * use Delta Kernel for reading Delta tables.
  */
-public class SparkScanBuilder implements ScanBuilder, SupportsPushDownRequiredColumns {
+public class SparkScanBuilder
+    implements ScanBuilder, SupportsPushDownRequiredColumns, SupportsPushDownFilters {
 
-  private final io.delta.kernel.ScanBuilder kernelScanBuilder;
+  private io.delta.kernel.ScanBuilder kernelScanBuilder;
   private final String tablePath;
   private final StructType dataSchema;
   private final StructType partitionSchema;
   private final CaseInsensitiveStringMap options;
   private final Set<String> partitionColumnSet;
   private StructType requiredDataSchema;
-  private Predicate[] pushedPredicates;
+  private Predicate[] pushedKernelPredicates;
```

Comment:
```
// pushedKernelPredicates: Predicates that have been pushed down to the Delta Kernel for evaluation.
// pushedFilters: The same pushed predicates, but represented using Spark’s {@link Filter} API
//                (needed because Spark operates on Filter objects while the Kernel uses Predicate)
```